### ISSUE TITLE
Support rolling policies in FileAppenderFactory

### ISSUE THREAD

**Issue by andrus** [nhl/bootique-logback#14]
```FileAppenderFactory``` should be expanded to support various rolling policies provided by Logback. Our initial "approximation" of the rolling policies has a few gaping holes that need to be plugged. The following properties of ```FileAppenderFactory``` will be removed: 

* ```rotate```
* ```maxFiles```
* ```maxTotalSize```
* ```maxFileSize```

replaced with ```rollingPolicy```.

**Comment by andrus** [comment]
@vvertinskiy : Merged #13. Improvement suggestions:

## RollingPolicyFactory refactoring
```java
public RollingPolicy createRollingPolicy(LoggerContext context) {
	FileNamePatternValidator.validate(fileNamePattern, context, getRollingPolicyType());
	return instantiatePolicy(context);
}
``` 
Let's make it more object-oriented. Instead of having a switch inside ```FileNamePatternValidator```, implement proper validators in subclasses of RollingPolicyFactory.


## Slow tests

LogbackBQConfigIT takes 71 seconds to finish. I understand that *some* of the tests are testing time-based rotation. But not all. Those that are not time-based should not suffer. Also tests for time-based rotation can probably do their thing in time periods shorter then 10 seconds.

**Comment by andrus** [comment]
closing (as 0.10 release is out... moved leftovers to #15)
